assets/src/application/ui/layout/EntityList/index.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { without } from 'ramda';
+import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+
+import { Entity } from '@edtrServices/apollo/types';
+import './style.scss';
+
+interface EntityListProps {
+	entities: Entity[];
+	EntityGridView: React.ReactNode;
+	EntityListView: React.ReactNode;
+	htmlClass: string;
+	view: string;
+	noResultsText: string;
+}
+
+/**
+ * EntityList
+ * base component for displaying a list of entities (dates, tickets, etc)
+ * as either a list table or grid of entity blocks
+ *
+ * @function
+ * @param {Array} entities
+ * @param {Component} EntityGridView
+ * @param {Component} EntityListView
+ * @param {string} htmlClass
+ * @param {string} view
+ * @param {string} noResultsText
+ * @param {mixed} otherProps
+ * @return {Component} list of rendered entities
+ */
+const EntityList = ({
+	entities = [],
+	EntityGridView,
+	EntityListView,
+	htmlClass = '',
+	view = 'grid',
+	noResultsText = '',
+	...otherProps
+}) => {
+	// verify array and remove undefined
+	const filteredEntities = Array.isArray(entities) ? without([undefined], entities) : [];
+
+	if (filteredEntities.length === 0) {
+		noResultsText = noResultsText !== '' ? noResultsText : __('no results found', 'event_espresso');
+		return <div className='ee-entity-list-no-results'>{noResultsText}</div>;
```

Comment:
this could maybe use https://ant.design/components/empty/ although that would likely require the styles to be changed

Reply:
Updated: https://github.com/eventespresso/event-espresso-core/pull/2366/commits/dfe12e3f8bb818b1675c02d46a5c09e5e356b742